tutorials/11-probabilistic-pca/11_probabilistic-pca.jmd

```diff
@@ -147,131 +164,79 @@ Note that:
- 4. then we assume uniform offset for each dimension $d \in D$, and draw samples from an MND for each dimension;
- 5. Finally, we iterate through each dimension in $D$, and define an MND (sampling distribution, likelihood), assuming all instances in that dimension follow an MND with mean $w \times z + mu_d$, for all instances in that dimension.
+ 2. draw samples of each entries of the projection matrix $\mathbf{W}$ from a standard normal;
+ 3. draw samples of the latent variable $\mathbf{Z}_{k \times N}$ from an MND;
+ 4. draw samples of the offset $\boldsymbol{\mu}$ from an MND, assuming uniform offset for all instances;
+ 5. Finally, we iterate through each gene dimension in $\mathbf{X}$, and define an MND for the sampling distribution (i.e. likelihood).
 
 #### Step 4: Sampling-based inference of the pPCA model
 
-Here we aim to perform MCMC sampling to infer the projection matrix $w_{D \times k}$, the latent variable matrix $z_{k \times N}$, and the offsets $mu_{D \times 1}$.
+Here we aim to perform MCMC sampling to infer the projection matrix $\mathbf{W}_{D \times k}$, the latent variable matrix $\mathbf{Z}_{k \times N}$, and the offsets $\boldsymbol{\mu}_{N \times 1}$.
 
-We run the inference using the NUTS sampler, of which the chain length is set to be 500. By default, the NUTS sampler samples 1 chain.
+We run the inference using the NUTS sampler, of which the chain length is set to be 500, target accept ratio 0.65 and initial stepsize 0.1. By default, the NUTS sampler samples 1 chain.
 You are free to try [different samplers](https://turing.ml/stable/docs/library/#samplers).
 
 ```julia
 k = 2 # k is the dimension of the projected space, i.e. the number of principal components/axes of choice
-ppca = pPCA(expression_matrix, k) # instantiate the probabilistic model
-chain_ppca = sample(ppca, NUTS(), 500);
+ppca = pPCA(Array(mat_exp), k) # ppca = pPCA_1(Array(df_exp), k), instantiate the probabilistic model
+chain_ppca = sample(ppca, NUTS(0.65; init_ϵ=0.1), 500);
 ```
 
-The sampling process takes about 7 mins. The samples are saved in the Chains struc `chain_ppca`, whose shape can be checked:
+The sampling process takes about 10 mins to run on a 4-core x86_64 computer. The samples are saved in the Chains struct `chain_ppca`, whose shape can be checked:
 
 ```julia
-size(chain_ppca)
+size(chain_ppca) # (no. of iterations, no. of vars, no. of chains) = (500, 159, 1)
 ```
 
-The Chains struc `chain_ppca` also contains the sampling info such as rhat, ess, mean estimates, etc. You can print it to check these quantities.
-
-#### Step 5: diagnostics and posterior predictive checks
-
-Diagnostics of the inference results, i.e. convergence (r-hat), effective samples size (ess), acceptance rate, etc, can be performed using the `MCMCChain` package.
-Here we plot the sample trajectories for 1 chain:
+The Chains struc `chain_ppca` also contains the sampling info such as r-hat, ess, mean estimates, etc. You can print it to check these quantities.
```

Comment:
I think it's a bit questionable whether this should be included in the tutorial.

Reply:
Maybe rather just remove the semicolon in the line where the chain is sampled, then everything will be displayed automatically.